pkg/shared/queue/overflow_queue.go

```diff
@@ -0,0 +1,56 @@
+package queue
+
+import "sync"
+
+// OverflowQeueue is a thread safe queue implementation with max size, and the oldest elements automatically overflow.
+type OverflowQueue[T any] struct {
+	elements []T
+	maxSize  int
+	lock     *sync.RWMutex
+}
+
+func New[T any](size int) *OverflowQueue[T] {
+	return &OverflowQueue[T]{
+		elements: []T{},
+		maxSize:  size,
+		lock:     new(sync.RWMutex),
+	}
+}
+
+// Append adds an element to the queue
+func (q *OverflowQueue[T]) Append(value T) {
+	q.lock.Lock()
+	defer q.lock.Unlock()
+	if len(q.elements) >= q.maxSize {
```

Comment:
NIT: likely path is going to run this branch all the time.

Reply:
Will evaluate the performance of using container.List later.